specs/agents/mobile/metrics.md

```diff
@@ -0,0 +1,41 @@
+## Mobile Metrics
+
+### CPU metrics
+| Name               | Type             | Units      | Description                     |
+|--------------------|------------------|------------|---------------------------------|
+| `system.cpu.usage` | Observable Guage | percentage | A percentage value of cpu usage |
+
+### Memory Metrics
+| Name                   | Type             | Units | Description                             |
+|------------------------|------------------|-------|-----------------------------------------|
+| `system.memory.usage`  | Observable Guage | bytes | The application's memory usage in bytes |
+
+
+### Application Metrics
+#### load times
+| Name                                          | Type      | Units   | Description                                                           |
+|-----------------------------------------------|-----------|---------|-----------------------------------------------------------------------|
+| `application.launch.optimizedTimeToFirstDraw` | histogram | seconds | The amount of time spent launching the app until loaded (optimized)   |
+| `application.launch.timeToFirstDraw`          | histogram | seconds | The amound of time spent launching the app until loaded               |
```

Comment:
My my own edification: what's the difference between these? Would it make sense to have one metric name, and a separate dimension which indicates whether the application is optimized (if that's what it means)?

Reply:
I'm not sure if in Android has an equivalent to "optimized time to first draw". We have cold, warm and hot launch types, where essentially only the "cold" one is the actual app launch, and the other ones are we users bring the app to the foreground after it was previously launched and then minimized, so I'm not sure if we can count them as app launch? Though anyway, I haven't gotten to this part yet in the Android agent so I'm not fully informed of what we can measure from any Android app, but I can share this link for now where they explain what I just mentioned more in detail, just in case it can be helpful to get an idea of what could be some common data across platforms: https://developer.android.com/topic/performance/vitals/launch-time

Reply:
According to the Apple documentation `OptimizedTimeToFirstDraw` is a prewarmed launch similar to the android launch time breakdown. Maybe we should used separate dimensions to describe launch timing similar to how I organized `application.exit` one for cold, warm, and hot? This would mean to fold `resumeTime` into this metric though... and that might be different than a "hot" start? https://developer.apple.com/documentation/metrickit/mxapplaunchmetric/3901512-histogrammedoptimizedtimetofirst